chain/consensus/mir/state_manager.go

```diff
@@ -685,9 +696,22 @@ func (sm *StateManager) releaseNextCheckpointChan() {
 	}
 }
 
+func (sm *StateManager) waitForHeightWithTimeout(height abi.ChainEpoch) error {
+	log.With("validator", sm.id).Infof("waitForHeight %v started", height)
+	defer log.With("validator", sm.id).Infof("waitForHeight %v finished", height)
+
+	ctx, cancel := context.WithDeadline(sm.ctx, time.Now().Add(5*time.Second))
```

Comment:
Why using `WIthDeadline` instead of `WithTimeout`? Also we may need more time, right? What if the gap is big, 5 seconds may not be enough, and we may need to sync a huge gap, we should account for that as we do in other places.

Reply:
Fixed. Agree

Reply:
30 seconds may still not be enough, can we make it dependent of the gap between the current height and the one we are targeting?

Reply:
Fixed by adding backoff